lib/galaxy/util/zipstream.py

```diff
@@ -41,7 +42,7 @@ def response(self) -> Iterator[bytes]:
     def get_headers(self) -> Dict[str, str]:
         headers = {}
         if self.archive_name:
-            headers["Content-Disposition"] = f'attachment; filename="{self.archive_name}.zip"'
+            headers["Content-Disposition"] = f'attachment; filename="{decodify(self.archive_name)}.zip"'
```

Comment:
Can we instead just do the inline call ? Looking at this I know exactly what it's going to do, looking at a function like decodify I don't.
```suggestion
            headers["Content-Disposition"] = f'attachment; filename="{self.archive_name.encode('latin-1', 'replace').decode('latin-1')}.zip"'
```
?

Reply:
Because this problem does not just belong to this specific purpose (#18584), we could use it in other places.

Reply:
We're fixing a specific bug here, with a very specific standard spec stating we need latin-1 characters in headers. This is not liekly to be the case for https://github.com/galaxyproject/galaxy/issues/18584. If it turns out that all of this is needed to accomplish https://github.com/galaxyproject/galaxy/issues/18584 I don't mind, but that should go to another PR.

Reply:
Sure, I will make this PR specific to the issue we are addressing.
However, why are we using 'latin-1' encoding? It might not display the correct characters when downloading.

Reply:
https://github.com/encode/starlette/issues/1098#issuecomment-866912809